pennylane/devices/default_qubit.py

```diff
@@ -577,17 +577,17 @@ def preprocess_transforms(
 
         if config.interface == math.Interface.JAX_JIT:
             transform_program.add_transform(no_counts)
-        transform_program.add_transform(
-            mid_circuit_measurements, device=self, mcm_config=config.mcm_config
-        )
-        # validate_device_wires needs to be after defer_measurement has added more wires.
-        transform_program.add_transform(validate_device_wires, self.wires, name=self.name)
         transform_program.add_transform(
             decompose,
             stopping_condition=stopping_condition,
             stopping_condition_shots=stopping_condition_shots,
             name=self.name,
         )
+        transform_program.add_transform(device_resolve_dynamic_wires, wires=self.wires)
+        transform_program.add_transform(
+            mid_circuit_measurements, device=self, mcm_config=config.mcm_config
+        )
```

Comment:
What if the deferred measurement transform added additional controlled operators that need to be further decomposed?